File: MasterErp.Angular/src/app/components/Inventory/models/ItemLookupModel.ts
import { CreatorModel } from "../../Shared/models/CreatorModel";

export interface ItemLookupModel extends CreatorModel {
    itemLookupId: number | null;
    nameAR: string;
    nameEN: string;
    notes: string;
    branchId: number | null;
    items:ItemLookupDetailsModel[]
}

export interface ItemLookupDetailsModel {
    itemLookupDetailsId: number | null;
    itemId: number;
    itemNameEN: string;
    itemNameAR: string;
    displayOrder: number | null;
    quantity: number | null;
}

File: RetailStation.Entities/DTOs/Operation/ItemLookupModel.cs
using RetailStation.Entities.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetailStation.Entities.DTOs.Operation
{

    public class ItemLookupModel : CreatorModel
    {
        public int? ItemLookupId { get; set; }
        public string NameAR { get; set; }
        public string NameEN { get; set; }
        public string Notes { get; set; }
        public int? BranchId { get; set; }
        public int? TotalCount { get; set; }
        public List<ItemLookupDetailsModel> Items { get; set; }
    }
    public class ItemLookupDetailsModel : ItemLookupModel
    {
        public int? ItemLookupDetailsId { get; set; }
        public int ItemId { get; set; }
        public string ItemNameEN { get; set; }
        public string ItemNameAR { get; set; }
        public string UnitNameEN { get; set; }
        public string UnitNameAR { get; set; }
        public int? DisplayOrder { get; set; }
        public double? Quantity { get; set; }
        public double? Price { get; set; }
    }
}
